include/seastar/core/function_traits.hh

```diff
@@ -57,6 +57,21 @@ template <typename T, typename Ret, typename... Args>
 struct function_traits<Ret(T::*)(Args...) const> : public function_traits<Ret(Args...)>
 {};
 
+#if __cplusplus >= 201703L
```

Comment:
You can drop this #ifdef, Seastar support C++20 and C++23 only.

Reply:
done

Reply:
@avikivity , is anything else required here?